File: 2316-count-hills-and-valleys-in-an-array/count-hills-and-valleys-in-an-array.py
class Solution:
    def countHillValley(self, nums: list[int]) -> int:
        count = 0
        left = 0  # Index of last non-equal value

        for i in range(1, len(nums) - 1):
            if nums[i] != nums[i + 1]:
                if (nums[i] > nums[left] and nums[i] > nums[i + 1]) or \
                   (nums[i] < nums[left] and nums[i] < nums[i + 1]):
                    count += 1
                left = i  # Update left to current index

        return count

File: 2316-count-hills-and-valleys-in-an-array/count-hills-and-valleys-in-an-array.go
func countHillValley(nums []int) int {
	count := 0
	left := 0 // Index of last non-equal element

	for i := 1; i < len(nums)-1; i++ {
		if nums[i] != nums[i+1] {
			if (nums[i] > nums[left] && nums[i] > nums[i+1]) ||
				(nums[i] < nums[left] && nums[i] < nums[i+1]) {
				count++
			}
			left = i // Update left for next round
		}
	}

	return count
}
